numpy/f2py/cfuncs.py

```diff
@@ -1227,8 +1234,8 @@
             strcpy(mess, "try_pyarr_from_character failed"
                          " -- expected bytes array-scalar|array, got ");
             f2py_describe(obj, mess + strlen(mess));
+            PyErr_SetString(err, mess);
         }
-        PyErr_SetString(err, mess);
```

Comment:
mess is uninitialized?  So better not use it at all.  The other places compose a new message outside the branch.  But I am not convinced that is even useful, so...